tests/Core/Tokenizer/EnumCaseTest.inc

```diff
@@ -0,0 +1,65 @@
+<?php
+
+enum Foo
+{
+    /* testPureEnumCase */
+    case SOME_CASE;
+}
+
+enum Boo: int {
+    /* testBackingIntegerEnumCase */
+    case ONE = 1;
+}
+
+enum Hoo: string
+{
+    /* testBackingStringEnumCase */
+    case ONE = 'one';
+}
+
+enum ComplexEnum: int implements SomeInterface
+{
+    use SomeTrait {
+        traitMethod as enumMethod;
+    }
+
+    const SOME_CONSTANT = true;
+
+    /* testEnumCaseInComplexEnum */
+    case ONE = 1;
+
+    /* testEnumCaseIsCaseInsensitive */
+    CaSe TWO = 2;
+
+    public function someMethod(): bool
+    {
+        switch (true) {
+            /* testCaseWithSemicolonIsNotEnumCase */
+            case CONSTANT;
+        }
+    }
+
+    /* testEnumCaseAfterSwitch */
+    case THREE = 3;
+
+    public function someOtherMethod(): bool
+    {
+        switch (true):
+            case false:
+        endswitch;
+    }
+
+    /* testEnumCaseAfterSwitchWithEndSwitch */
+    case FOUR = 4;
+}
+
+switch (true) {
+    /* testCaseWithConstantIsNotEnumCase */
+    case CONSTANT:
+    /* testCaseWithConstantAndIdenticalIsNotEnumCase */
+    case CONSTANT === 1:
+    /* testCaseWithAssigmentToConstantIsNotEnumCase */
+    case CONSTANT = 1:
+    /* testIsNotEnumCaseIsCaseInsensitive */
+    cAsE CONSTANT:
+}
```

Comment:
I'd recommend adding far more extensive tests for varying combinations of `enum` and `switch`, as well as covering as many situations as one can think of where `case` can be used and should be tokenized as `T_STRING` instead of `T_CASE` or `T_ENUM_CASE`.

See the test cases for the `DefaultKeywordTest` for inspiration.